### PR TITLE
[receiver/mysql]: add mysql.client.network.io metric

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

add mysql.bytes.count metric

**Link to tracking Issue:**

#14138

**Testing:**

unit tests

**Documentation:**

`metadata.yaml`

### CODE REVIEW

File: receiver/mysqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -283,6 +287,16 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [threads]
+  mysql.bytes.count:
+    enabled: true
+    description: The number of transmitted bytes.
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:292]: What kind of transmission is this? Network? To/from disk? I think this should be clear in the description and probably reflected in the name too.

File: receiver/mysqlreceiver/metadata.yaml [hunk 2]
```diff
@@ -283,6 +287,16 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [threads]
+  mysql.bytes.count:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:290]: Typically we should not use the unit in the name. Instead, I think `throughput` is usually appropriate.

File: receiver/mysqlreceiver/metadata.yaml [hunk 3]
```diff
@@ -283,6 +287,16 @@ metrics:
...
+    enabled: true
+    description: The number of transmitted bytes.
+    unit: 1
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:293]: Presumably this is `By`

File: receiver/mysqlreceiver/metadata.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml:292]: This is client transmission, so network

Comment (djaglowski) [reply receiver/mysqlreceiver/metadata.yaml:292]: 👍 I think it should end with `network.io` then, maybe `mysql.client.network.io`?

File: receiver/mysqlreceiver/metadata.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml]: Changed to `mysql.client_throughput.count`

Comment (djaglowski) [reply receiver/mysqlreceiver/metadata.yaml]: I believe I erred with my suggestion of using `throughput`. This type of metric is quite specifically called out in the naming guidelines as a [`.io` metric](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-naming)?

File: receiver/mysqlreceiver/metadata.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml]: changed to `mysql.client.network.io.count`, not sure if `count` should be applied here, I can strip it

Comment (djaglowski) [reply receiver/mysqlreceiver/metadata.yaml]: Since the semantic conventions specifically call out `.io` metrics, I think `count` can be omitted.

Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml]: removed and rebased

**Review by djaglowski — APPROVED** [review]

File: receiver/mysqlreceiver/metadata.yaml [hunk 4]
```diff
@@ -75,6 +75,10 @@ attributes:
   index_name:
     value: index
     description: The name of the index.
+  transmission_direction:
+    value: kind
```
[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml:79]: Fixed in #15366

[the remark this answers is not in the document]
Comment (djaglowski) [reply receiver/mysqlreceiver/metadata.yaml:79]: 👍 , I merged #15366. I think this PR can be closed now?

**Comment by djaglowski** [comment]
Closed based on #15366

**Pull request closed without merging**